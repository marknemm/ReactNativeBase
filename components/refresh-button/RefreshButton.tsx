import { REFRESH_ICON } from '@constants/icons';
import { FixedIconButtonProps } from '@interfaces/button';
import { Button, useTheme } from '@rneui/themed';
import { forwardRef } from 'react';

/**
 * The {@link RefreshButton} component.
 *
 * @param props The component {@link FixedIconButtonProps}.
 * @param ref The component reference.
 * @returns The {@link RefreshButton} component.
 */
const RefreshButton: React.FC<FixedIconButtonProps> = forwardRef(({
  color = 'transparent',
  iconColor,
  iconSize = 24,
  ...buttonProps
}, ref) => {
  const { theme } = useTheme();

  return (
    <Button
      accessibilityLabel="Refresh"
      color={color}
      icon={{
        ...REFRESH_ICON,
        color: iconColor ?? theme.colors.black,
        size: iconSize,
      }}
      {...buttonProps}
      ref={ref}
    />
  );
});

export default RefreshButton;
