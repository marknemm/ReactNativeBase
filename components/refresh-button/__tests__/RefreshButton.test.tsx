import RefreshButton from '@components/refresh-button/RefreshButton';
import AppProvider from '@test/contexts/app/AppProvider';
import { fireEvent, render, screen } from '@testing-library/react-native';

describe('<RefreshButton />', () => {
  it('calls onPress when the button is pressed', () => {
    const onPress = jest.fn();
    render(
      <RefreshButton onPress={onPress} />,
      { wrapper: AppProvider }
    );

    fireEvent.press(screen.getByRole('button'));
    expect(onPress).toHaveBeenCalled();
  });

  it('does not call onPress when disabled', () => {
    const onPress = jest.fn();
    render(
      <RefreshButton onPress={onPress} disabled />,
      { wrapper: AppProvider }
    );

    fireEvent.press(screen.getByRole('button'));
    expect(onPress).not.toHaveBeenCalled();
  });

  describe('snapshots', () => {
    it('renders correctly', () => {
      render(
        <RefreshButton />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly when disabled', () => {
      render(
        <RefreshButton disabled />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });
  });
});
