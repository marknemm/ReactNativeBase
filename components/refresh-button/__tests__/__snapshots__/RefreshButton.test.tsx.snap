// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RefreshButton /> snapshots renders correctly 1`] = `
<View
  style={
    [
      {
        "overflow": "hidden",
      },
      {
        "borderRadius": 3.142857142857143,
      },
      undefined,
      false,
    ]
  }
  testID="RNE_BUTTON_WRAPPER"
>
  <View
    accessibilityLabel="Refresh"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": false,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#007aff",
          "borderRadius": 3.142857142857143,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 12.571428571428571,
          "paddingHorizontal": 14.571428571428571,
        }
      }
    >
      <View
        style={
          {
            "marginHorizontal": 5,
            "overflow": "hidden",
          }
        }
        testID="RNE__ICON__CONTAINER"
      >
        <View
          accessibilityRole="button"
          android_ripple={
            {
              "borderless": false,
              "color": "rgba(0, 0, 0, 0.3)",
              "radius": -5,
            }
          }
          disabled={false}
          testID="RNE__ICON__CONTAINER_ACTION"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            testID="RNE__ICON"
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#242424",
                    "fontSize": 24,
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                  {
                    "fontFamily": "material",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="RNE__ICON__Component"
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<RefreshButton /> snapshots renders correctly when disabled 1`] = `
<View
  style={
    [
      {
        "overflow": "hidden",
      },
      {
        "borderRadius": 3.142857142857143,
      },
      undefined,
      false,
    ]
  }
  testID="RNE_BUTTON_WRAPPER"
>
  <View
    accessibilityLabel="Refresh"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": false,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "hsl(208, 8%, 90%)",
          "borderColor": "#007aff",
          "borderRadius": 3.142857142857143,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 12.571428571428571,
          "paddingHorizontal": 14.571428571428571,
        }
      }
    >
      <View
        style={
          {
            "marginHorizontal": 5,
            "overflow": "hidden",
          }
        }
        testID="RNE__ICON__CONTAINER"
      >
        <View
          accessibilityRole="button"
          android_ripple={
            {
              "borderless": false,
              "color": "rgba(0, 0, 0, 0.3)",
              "radius": -5,
            }
          }
          disabled={false}
          testID="RNE__ICON__CONTAINER_ACTION"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            testID="RNE__ICON"
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#242424",
                    "fontSize": 24,
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                  {
                    "fontFamily": "material",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="RNE__ICON__Component"
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
